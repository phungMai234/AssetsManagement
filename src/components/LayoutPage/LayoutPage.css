body {
  margin: 0;
  font-size: 14px;
  font-family: Helvetica Neue, Helvetica, Arial, sans-serif;
  overflow: auto;
}
* {
  box-sizing: border-box;
}
.text-center {
  align-content: center;
}
.header {
  height: 80px;
  padding: 15px;
  position: relative;
  display: flex;
  justify-content: space-between;
  align-items: center;
  align-content: center;
  background-color: #ffffff;
  border-bottom: 1px solid #eee;
}
.sidenav {
  top: 80px;
  width: 250px;
  bottom: 0;
  left: 0;
  position: absolute;
  background-color: #fff;
  padding-top: 10px;
  border-right: 1px solid #eee;
}
.content {
  top: 80px;
  position: absolute;
  left: 250px;
  bottom: 0;
  right: 0;
  padding: 10px;
  font-size: 14px;
}
