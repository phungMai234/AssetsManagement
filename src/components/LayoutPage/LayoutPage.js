import React from 'react';
import 'bootstrap/dist/css/bootstrap.css';
import 'bootstrap/dist/js/bootstrap.bundle';
import 'jquery';

import Header from '../header/header';
import Menu from '../menu/menu';
import MessageRes from 'components/MessageRes';

import { Row, Col } from 'react-bootstrap';
import useAlert from 'hooks/useAlert';
import Wrapper from './LayoutPage.styles';

function LayoutPage({ children }) {
  const { alert, clearAlert } = useAlert();

  return (
    <Wrapper>
      <div className="header">
        <Header />
      </div>
      <Row>
        <Col md={3} className="sidenav">
          <Menu />
        </Col>
        <Col md={9} className="content">
          <main>
            <Row>
              <Col md={12}>
                {!!alert && <MessageRes content={alert.message} status={alert.status} onHide={clearAlert} />}
              </Col>
            </Row>
            {children}
          </main>
        </Col>
      </Row>
    </Wrapper>
  );
}

export default LayoutPage;
