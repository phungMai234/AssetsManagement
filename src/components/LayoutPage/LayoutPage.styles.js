import styled from 'styled-components';

export default styled.div`
  /* body {
    margin: 0;
    font-size: 14px;
    font-family: Helvetica Neue, Helvetica, Arial, sans-serif;
    overflow: auto;
  } */
  /* * {
    box-sizing: border-box;
  } */
  .text-center {
    align-content: center;
  }
  .header {
    height: 80px;
    padding: 15px;
    position: relative;
    display: flex;
    justify-content: space-between;
    align-items: center;
    align-content: center;
    background-color: #ffffff;
    border-bottom: 1px solid #eee;
  }
  .sidenav {
    border-right: 1px solid #eee;
    padding: unset;
  }
  .content {
    padding: unset;
  }
`;
