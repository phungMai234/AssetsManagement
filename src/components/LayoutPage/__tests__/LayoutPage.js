import React from 'react';

import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import LayoutPage from '../LayoutPage';
import MessageContext from 'contexts/message';

const BaseComponent = ({ ...props }) => (
  <Router>
    <MessageContext.Provider value={{ alert: {}, clearAlert: jest.fn(), setAlert: jest.fn(), ...props }}>
      <LayoutPage />
    </MessageContext.Provider>
  </Router>
);

describe('Layout Page', () => {
  it('should render correctly', () => {
    const container = render(<BaseComponent>hello</BaseComponent>);
    expect(container.container).toMatchSnapshot();
  });
  it('should render correctly with alert message', () => {
    const container = render(<BaseComponent alert={{ message: 'test', status: 'success' }}>hello</BaseComponent>);
    expect(container.container).toMatchSnapshot();
  });
});
