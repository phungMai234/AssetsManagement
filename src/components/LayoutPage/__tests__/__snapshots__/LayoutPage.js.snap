// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Page should render correctly 1`] = `
<div>
  <div>
    <div
      class="header"
    >
      <a
        class="header-left "
        href="/"
      >
        <img
          alt="logo"
          src="https://firebasestorage.googleapis.com/v0/b/assetsmanagementfirebase.appspot.com/o/images%2Flogo2_new.png?alt=media&token=338cbb32-2c47-4945-bfcd-081bf460eb54"
        />
        <span
          class="logo-txt"
        >
          UET Assets Management
        </span>
      </a>
      <div
        class="header-right"
      >
        <div
          class="dropdown"
        >
          <button
            aria-expanded="false"
            aria-haspopup="true"
            class="dropdown-toggle btn btn-primary btn-sm"
            id="dropdown-menu-align-right"
            type="button"
          >
            <span>
              <svg
                fill="none"
                height="15"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
                />
                <circle
                  cx="12"
                  cy="7"
                  r="4"
                />
              </svg>
              Admin
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="sidenav"
    >
      <ul
        class="nav flex-column nav-pills nav-fill nav-menu"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link "
            href="/dashboard/categories"
          >
            <svg
              fill="none"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 9l9-7 9 7v11a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
              />
              <polyline
                points="9 22 9 12 15 12 15 22"
              />
            </svg>
            <span>
              Quản lý các loại tài sản
            </span>
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link "
            href="/dashboard/assets"
          >
            <svg
              fill="none"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <ellipse
                cx="12"
                cy="5"
                rx="9"
                ry="3"
              />
              <path
                d="M21 12c0 1.66-4 3-9 3s-9-1.34-9-3"
              />
              <path
                d="M3 5v14c0 1.66 4 3 9 3s9-1.34 9-3V5"
              />
            </svg>
            <span>
              Quản lý tài sản
            </span>
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link "
            href="/dashboard/delivery_reports"
          >
            <svg
              fill="none"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V8z"
              />
              <polyline
                points="14 2 14 8 20 8"
              />
              <line
                x1="16"
                x2="8"
                y1="13"
                y2="13"
              />
              <line
                x1="16"
                x2="8"
                y1="17"
                y2="17"
              />
              <polyline
                points="10 9 9 9 8 9"
              />
            </svg>
            <span>
              Quản lý biên bản bàn giao
            </span>
          </a>
        </li>
      </ul>
    </div>
    <div
      class="content"
    >
      <main>
        <div
          class="row"
        >
          <div
            class="col-md-12"
          >
            <div
              class="sc-bdfBwQ fIHKji alert alert-undefined alert-dismissible fade show"
              role="alert"
            >
              <button
                aria-label="Close"
                class="close"
                data-dismiss="alert"
                style="width: auto;"
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  ×
                </span>
              </button>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;

exports[`Layout Page should render correctly with alert message 1`] = `
<div>
  <div>
    <div
      class="header"
    >
      <a
        class="header-left "
        href="/"
      >
        <img
          alt="logo"
          src="https://firebasestorage.googleapis.com/v0/b/assetsmanagementfirebase.appspot.com/o/images%2Flogo2_new.png?alt=media&token=338cbb32-2c47-4945-bfcd-081bf460eb54"
        />
        <span
          class="logo-txt"
        >
          UET Assets Management
        </span>
      </a>
      <div
        class="header-right"
      >
        <div
          class="dropdown"
        >
          <button
            aria-expanded="false"
            aria-haspopup="true"
            class="dropdown-toggle btn btn-primary btn-sm"
            id="dropdown-menu-align-right"
            type="button"
          >
            <span>
              <svg
                fill="none"
                height="15"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
                />
                <circle
                  cx="12"
                  cy="7"
                  r="4"
                />
              </svg>
              Admin
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="sidenav"
    >
      <ul
        class="nav flex-column nav-pills nav-fill nav-menu"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link "
            href="/dashboard/categories"
          >
            <svg
              fill="none"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 9l9-7 9 7v11a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
              />
              <polyline
                points="9 22 9 12 15 12 15 22"
              />
            </svg>
            <span>
              Quản lý các loại tài sản
            </span>
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link "
            href="/dashboard/assets"
          >
            <svg
              fill="none"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <ellipse
                cx="12"
                cy="5"
                rx="9"
                ry="3"
              />
              <path
                d="M21 12c0 1.66-4 3-9 3s-9-1.34-9-3"
              />
              <path
                d="M3 5v14c0 1.66 4 3 9 3s9-1.34 9-3V5"
              />
            </svg>
            <span>
              Quản lý tài sản
            </span>
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link "
            href="/dashboard/delivery_reports"
          >
            <svg
              fill="none"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V8z"
              />
              <polyline
                points="14 2 14 8 20 8"
              />
              <line
                x1="16"
                x2="8"
                y1="13"
                y2="13"
              />
              <line
                x1="16"
                x2="8"
                y1="17"
                y2="17"
              />
              <polyline
                points="10 9 9 9 8 9"
              />
            </svg>
            <span>
              Quản lý biên bản bàn giao
            </span>
          </a>
        </li>
      </ul>
    </div>
    <div
      class="content"
    >
      <main>
        <div
          class="row"
        >
          <div
            class="col-md-12"
          >
            <div
              class="sc-bdfBwQ fIHKji alert alert-success alert-dismissible fade show"
              role="alert"
            >
              test
              <button
                aria-label="Close"
                class="close"
                data-dismiss="alert"
                style="width: auto;"
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  ×
                </span>
              </button>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
